feat: additionalClassSchema (#139)
vuelidate/formvuelatte: index.md

  <<< .vitepress/docs/components/HorizontalForm.vue
</SplitTab>

### Prop: schemaRowClasses

If you ever require to add additional classes to each structural row in the generated form, you can use the `schemaRowClasses` property of the `SchemaForm` component to inject them.

These classes will be appended, and will _not_ substitute the `schema-row` class that FormVueLate already provides.

The `schemaRowClasses` prop accepts `[String, Object, Array]` as valid types, conforming to Vue's [class and style binding syntax](https://v3.vuejs.org/guide/class-and-style.html#binding-html-classes)

```html
<SchemaForm
  schemaRowClasses="my-custom-class-a my-custom-class-b"
  :schema="mySchema"
/>
```

Example output:

```html
<form>
  <div class="schema-row custom-class-a custom-class-b">
    [...]
  </div>
</form>

> ```

Reviewer: Could you please also add a space after this bit?

Author: Done. I love your attention to this details. Have to improve personally for this.

  <<< .vitepress/docs/components/HorizontalForm.vue
</SplitTab>

### Prop: schemaRowClasses

> If you ever require to add additional classes to each structural row in the generated form, you can use the `schemaRowClasses` property to inject them.

Reviewer: ```suggestion
If you ever require to add additional classes to each structural row in the generated form, you can use the `schemaRowClasses` property of the `SchemaForm` component to inject them.
```

These classes will be appended, and will _not_ substitute the `schema-row` class that FormVueLate already provides.

> The prop accepts `[String, Object, Array]` as valid types.

Reviewer: ```suggestion
The `schemaRowClasses` prop accepts `[String, Object, Array]` as valid types, conforming to Vue's [class and style binding syntax](https://v3.vuejs.org/guide/class-and-style.html#binding-html-classes) 
```

> The philosophy is `Bring Your Own Component` and there will be moments when you already have your own css or a Framework UI you are using dictates that you need to use specific classes.
> 
> For those opinated situations `schemaRowClasses` allows you to complement the `schema-row` class that Formvuelate provides.
> 
> The prop accepts `[String, Object, Array]`, since any of those are valid ways to set CSS in Vue.

Reviewer: ```suggestion
If you ever require to add additional classes to each structural row in the generated form, you can use the `schemaRowClasses` property to inject them.

These classes will be appended, and will _not_ substitute the `schema-row` class that FormVueLate already provides.

The prop accepts `[String, Object, Array]` as valid types.
```

```html
<SchemaForm

>   :schemaRowClasses="my-custom-class-a my-custom-class-b"

Reviewer: ```suggestion
  schemaRowClasses="my-custom-class-a my-custom-class-b"
```

  :schema="mySchema"
/>
```

> Example output:

Reviewer: Please add a space between the closing 3 quotes in 246 and the next line

```html
<form>
  <div class="schema-row custom-class-a custom-class-b">

>     Fields

Reviewer: ```suggestion
    [...]
```

> When the `Form` html element is rendered, there will be a `div` element with those classes:

Reviewer: ```suggestion
Example output:
```

  </div>
</form>
```
:::tip
You might find scenarios where the Field must have a class that depends on the class applied to `schemaRowClasses`. Fortunately, Formvuelate has you covered.

For instance if you need this:

> ```html

Reviewer: This whole bit with the tip until the end is nice, but I feel it overexplains component composition which is not directly related to what FVL does. Let's remove this bit and keep it short and clean

> :::tip
> You might find scenarios where the Field must have a class that depends on the class applied to `schemaRowClasses`. Fortunately, Formvuelate has you covered.
> 
> For instance if you need this:
> ```html
> <form>
>   <div class="schema-row custom-class-a custom-class-b">
>     <input class="dependant-custom-class-a">
>   </div>
> </form>
> ```
> Your schema and component need to use the Prop class
> 
> ```html
> <template>
>   <SchemaForm :schema="schema" v-model="formData"
>   :schemaRowClasses="my-custom-class-a my-custom-class-b" />
> </template>
> 
> <script>
>   import { SchemaForm } from 'formvuelate'
>   import FormText from 'path/to/FormText'
>   import { ref } from 'vue'
> 
>   export default {
>     components: { SchemaForm },
>     setup() {
>       const schema = ref({
>         name: {
>           component: FormText,
>           class:'dependant-custom-class-a'
>       })
>       const formData = ref({})
> 
>       return {
>         schema,
>         formData
>       }
>     }
>   }
> </script>
> ```
> :::
> 
> Your component (in this example `FormText`) must have the prop class and use it.
> 

Reviewer: ```suggestion
```
